run.c

```diff
@@ -213,8 +218,7 @@ void run(void)
         /* more steps to go. */
 
         /* assign new timesteps to the active particles, now that we know they have synched TiKick and TiDrift */
-
-        find_timesteps();
+        minTimeBin = find_timesteps();
```

Comment:
I suggest using ``int find_timesteps(int * MinTimeBin)``; in the longer term we shall probably make all functions like this, because it makes it mentally easier to add more output parameters.